app.js

```diff
@@ -1,12 +1,37 @@
 const express = require("express");
 const app = express();
 app.use(express.json());
-const { getTopics } = require("./controllers/topic-controller");
+const {
+  getTopics,
+  getArticles,
+  getArticleById,
+} = require("./controllers/topic-controller");
 
 app.get("/api/topics", getTopics);
 
-app.use((request, response, next) => {
+app.get("/api/articles", getArticles);
+
+app.get("/api/articles/:article_id", getArticleById);
+
+app.use((err, request, response, next) => {
+  if (err.code === "22P02") {
+    response
+      .status(400)
+      .send({ status: 400, message: "article id must be a number" });
+  } else {
+    next(err);
+  }
+});
 
+app.use((err, request, response, next) => {
+  if (err.status && err.message) {
+    response.status(err.status).send(err);
+  } else {
+    next(err);
+  }
+});
+
+app.use((request, response, next) => {
```

Comment:
This middle function can go at the end of the endpoints and before the errors. I feel this placement makes more sense as it isn't strictly an error handling middleware function